scrapely/__init__.py

```diff
@@ -33,11 +36,12 @@ def train_from_htmlpage(self, htmlpage, data):
         assert data, "Cannot train with empty data"
         tm = TemplateMaker(htmlpage)
         for field, values in data.items():
-            if not hasattr(values, '__iter__'):
+            if (isinstance(values, six.string_types) or
+                    not hasattr(values, '__iter__')):
                 values = [values]
             for value in values:
-                if isinstance(value, str):
-                    value = value.decode(htmlpage.encoding or 'utf-8')
+                if isinstance(value, six.string_types):
+                    value = str_to_unicode(value, htmlpage.encoding)
```

Comment:
I think we can avoid the indirection and make code more straightforward by replacing 'str' with 'bytes':

```
if isinstance(value, bytes):
    value = value.decode(htmlpage.encoding or 'utf8')
```